src/elements/content-uploader/ContentUploader.tsx

```diff
@@ -994,10 +990,13 @@ class ContentUploader extends Component<ContentUploaderProps, State> {
         };
 
         if (items.length === 0) {
+            this.itemIdsRef.current = {};
             state.itemIds = {};
             state.errorCode = '';
         }
 
+        this.itemsRef.current = items;
+
         this.setState(state as Pick<State, keyof State>, callback);
```

Comment:
If we surrounded this call with flushSync would it solve a lot of our problems?

Reply:
no unfortunately. That was the first thing I tried